### PR TITLE
Implement rudimentary fulltext query expansion

### PR DESCRIPTION
**Pull request by rnewman**
This builds SQL for the simplest `fulltext` queries. Waddaya think, @ncalexan?

### CODE REVIEW

File: src/datomish/query/source.cljc [hunk 1]
```diff
@@ -35,13 +35,17 @@
 (defprotocol Source
   (source->from [source attribute]
     "Returns a pair, `[table alias]` for a pattern with the provided attribute.")
+  (source->non-fts-from [source])
```
Comment (ncalexan) [inline src/datomish/query/source.cljc]: Would you mind always using `fulltext` in names?  Might as well make it easy to grep.

File: src/datomish/query/source.cljc [hunk 2]
```diff
@@ -72,6 +76,14 @@
             (:fts-view source))]
       [table ((:table-alias source) table)]))
 
+  (source->non-fts-from [source]
```
Comment (ncalexan) [inline src/datomish/query/source.cljc]: I was thinking, late last night, about special representations for certain data types.  Is it worth trying to generalize a little here?  For example, we might want to add value hashes to certain rows, in order to have fast equality comparisons for large values; or we might want to represent URLs interned (like fulltext, but perhaps stored in some other representation).

File: src/datomish/query/functions.cljc [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+    (when-not (fulltext-attribute? (:source cc) (:value attr))
+      (raise-str "Attribute " (:value attr) " is not a fulltext-indexed attribute."))
+
+    (when-not (and (instance? BindColl (:binding function))
+                   (instance? BindTuple (:binding (:binding function)))
+                   (every? (fn [s]
+                             (or (instance? BindIgnore s)
+                                 (and (instance? BindScalar s)
+                                      (instance? Variable (:variable s)))))
+                           (:bindings (:binding (:binding function)))))
```
Comment (ncalexan) [inline src/datomish/query/functions.cljc]: This is pretty hard to read :)

**Comment by ncalexan** [comment]
@rnewman this is looking really good!  It would be nice to simplify some of the DS parser destructuring, but this is surprisingly easy for me to follow.

**Comment by rnewman** [comment]
Landed in b1e6ffc with review comments addressed.

**Pull request closed without merging**